kube-client/src/client/client_ext.rs

```diff
@@ -42,9 +45,90 @@ pub struct Cluster;
 /// You can create this directly, or convert `From` a `String` / `&str`, or `TryFrom` an `k8s_openapi::api::core::v1::Namespace`
 pub struct Namespace(String);
 
+/// Referenced object name resolution
+pub trait ObjectRef<K>: ObjectUrl<K> {
+    fn name(&self) -> Option<&str>;
+}
+
+/// Reference resolver for a specified namespace
+pub trait NamespacedRef<K>: ObjectUrl<K> {
+    /// Resolve reference in the provided namespace
+    fn within(&self, namespace: String) -> impl ObjectRef<K>;
+}
```

Comment:
I guess this will be a new required trait to use for people doing dynamic object ref -> object url inferences.

I think this is a good solution, but probably one that makes us perhaps want to consider a prelude down the road.